trino/sqlalchemy.py

```diff
@@ -0,0 +1,166 @@
+    driver = 'rest'
+    preparer = TrinoIdentifierPreparer
+    statement_compiler = TrinoCompiler
+    supports_alter = False
+    supports_pk_autoincrement = False
+    supports_default_values = False
+    supports_empty_insert = False
+    supports_multivalues_insert = True
+    supports_unicode_statements = True
+    supports_unicode_binds = True
+    returns_unicode_strings = True
+    description_encoding = None
+    supports_native_boolean = True
+    type_compiler = TrinoTypeCompiler
+
+    @staticmethod
+    def dbapi():
+        return trino.dbapi
+
+    @staticmethod
+    def create_connect_args(url):
+        """Construct args for Connection from SQLAlchemy connection string"""
+        kwargs = {
+            'user': url.username,
+            'host': url.host,
+            'port': url.port or 8080,
+        }
+
+        if url.query.get('ssl') == 'true':
+            kwargs['http_scheme'] = trino.constants.HTTPS
+
+        if url.password is not None:
+            kwargs['auth'] = trino.auth.BasicAuthentication(url.username, url.password)
+
+        if url.database is not None:
+            database_parts = url.database.split('/')
+            if len(database_parts) == 1:
+                kwargs['catalog'] = database_parts[0]
+            elif len(database_parts) == 2:
+                kwargs['catalog'] = database_parts[0]
+                kwargs['schema'] = database_parts[1]
+            else:
+                raise ValueError('Unexpected database format {}'.format(url.database))
+
+        return [], kwargs
+
+    def _get_table_columns(self, connection, table_name, schema):
+        """Helper to determine whether table with schema exists"""
+        full_table = self.identifier_preparer.quote_identifier(table_name)
+        if schema:
+            full_table = self.identifier_preparer.quote_identifier(schema) + '.' + full_table
+        try:
+            return connection.execute('SHOW COLUMNS FROM {}'.format(full_table))
+        except (trino.DatabaseError, exc.DatabaseError) as e:
+            msg = (
+                e.args[0].get('message') if e.args and isinstance(e.args[0], dict)
+                else e.args[0] if e.args and isinstance(e.args[0], str)
+                else None
+            )
+            regex = r'Table\ \'.*{}\'\ does\ not\ exist'.format(re.escape(table_name))
+            if msg and re.search(regex, msg):
+                raise exc.NoSuchTableError(table_name)
+            else:
+                raise
+
+    def has_table(self, connection, table_name, schema=None):
+        """Determine whether table with schema exists"""
+        try:
+            self._get_table_columns(connection, table_name, schema)
+            return True
+        except exc.NoSuchTableError:
+            return False
+
+    def get_indexes(self, connection, table_name, schema=None, **kw):
+        """Return columns which are partition keys"""
+        rows = self._get_table_columns(connection, table_name, schema)
```

Comment:
Should get columns from `<table>$partitions`. See [Hive special tables](https://trino.io/docs/current/connector/hive.html#special-tables)